f3ast/stream_builder.py

```diff
@@ -128,3 +127,35 @@ def split_dwells(dwells, max_dwt):
         dwells_reduced = dwells.copy()
         dwells_reduced[:, 0] = dwells_reduced[:, 0] / n_splits
         return [dwells_reduced for i in range(n_splits)]
+
+
+def stream_exp_correction_with_z(struct, settings, GR0=40e-3, doubling_length=500., sigma=4.2):
+    """ 
+    Returns stream model with exponential correction over structure height.
+    Might be useful for stl files with disconnected components, which leads to a breakdown of the DDModel.
+    
+    Takes initial growth rate/time from GR, and doubles dwell time over the length scale of doubling_length.    Doubling_length needs to be determined experimentally, e.g., from pitch of periodic structures over heights.
+    
```

Comment:
Please fix the spacing and too long lines errors. I suggest using `autopep8` and `pycodestyle` to check for these things.
There are a few code style fixes that I need to introduce in general. I plan to enforce correct formatting in the future, but haven't had the time to do it yet.